### PR TITLE
[KHRGA-38] Updated EP group class

### PR DESCRIPTION
**Pull request by imvucic**
Issue: KHRGA-38

### CODE REVIEW

File: source/iface/group.rst [hunk 1]
```diff
@@ -4,114 +4,316 @@
...
+  template <typename DataT>
+  device_event async_work_group_copy(global_ptr<DataT> dest,
+                                     local_ptr<DataT> src,
+                                     size_t numElements, size_t destStride) const
+
+All four ``async_work_group_copy`` functions listed above are deprecated
+in SYCL 2020. Has the same effect as the overload taking
+``decorated_local_ptr`` and ``decorated_global_ptr``
+except that the ``dest`` and ``src`` parameters are
+``multi_ptr#s with [code]#access::decorated::legacy``.
```
Comment (mkinsner) [inline source/iface/group.rst:281]: Formatting is broken here

Comment (imvucic) [reply source/iface/group.rst]: Removed 's' with https://github.com/KhronosGroup/SYCL_Reference/pull/116/commits/03bb586a2631aa3ede22960fc77267458f093781

Comment (mkinsner) [reply source/iface/group.rst]: This is still rendering with markup in it.  To clarify, the text in SYCL Reference currently renders as:

except that the `dest` and `src` parameters are `multi_ptr# with [code]#access::decorated::legacy`

The markup from the SYCL specification should not be included in the SYCL reference.  When the [spec fix PR](https://github.com/KhronosGroup/SYCL-Docs/pull/518) is merged, the SYCL specification will no longer show the markup such as `[code]` and `#`.  Those markup tags don't belong in the rendered text.

So for example, the above clip in the SYCL reference should instead look like:

except that the `dest` and `src` parameters are `multi_ptr` with `access::decorated::legacy`

Comment (imvucic) [reply source/iface/group.rst]: Thanks for detailed description...fixed with https://github.com/KhronosGroup/SYCL_Reference/pull/116/commits/d043aec22cb0671b66095bb468712c9cbfba2716

File: source/iface/group.rst [hunk 2]
```diff
@@ -4,114 +4,316 @@
...
+a particular work-group within a parallel execution.
+It is not user-constructible.
+
+The local range stored in the group class is provided either by the
+programmer, when it is passed as an optional parameter to
+``parallel_for_work_group``, or by the runtime system when it
+selects the optimal work-group size. This allows the developer
+to always know how many work-items are in each executing work-group,
+even through the abstracted iteration range
+of the ``parallel_for_work_item`` loops.
```
[the remark this answers is not in the document]
Comment (imvucic) [reply source/iface/group.rst]: fixed with https://github.com/KhronosGroup/SYCL_Reference/pull/116/commits/f017e97d83bf10144fe4f056a7718a04beffb5d5

**Review by mkinsner — APPROVED** [review]

**Pull request merged**